### PR TITLE
Add citation and annotation support for completion responses

### PR DESCRIPTION
**Pull request by mcowger**
## Summary
This PR adds support for processing citation and annotation fields in completion responses, enabling providers like Perplexity that include citation URLs to have those preserved through the transformation pipeline.
### Features
- **Citations support** : Add Citations field to OpenAI Response, store in TransformerMetadata, aggregate from streaming chunks
- **Annotations support**: Add Annotation/URLCitation structs, convert during transformations, aggregate with deduplication

- Add consistent nil/empty checks
- Add negative tests for invalid annotation data
## Testing
All tests pass including new tests for citations, annotations, and edge cases."

## Summary by CodeRabbit

* **New Features**
  * Messages now support citation annotations with URL metadata for referencing sources.
  * Responses include a list of citations for end-user reference.
  * Annotations are aggregated and deduplicated across streaming message chunks.

* **Tests**
  * Added comprehensive tests covering annotation handling, citation extraction, aggregation, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: llm/transformer/openai/aggregator.go [hunk 1]
```diff
@@ -192,6 +218,14 @@ func AggregateStreamChunks(ctx context.Context, chunks []*httpclient.StreamEvent
 			message.ToolCalls = finalToolCalls
 		}
 
+		// Set annotations if available
+		if len(choiceAgg.annotations) > 0 {
+			message.Annotations = make([]llm.Annotation, 0, len(choiceAgg.annotations))
+			for _, annotation := range choiceAgg.annotations {
+				message.Annotations = append(message.Annotations, annotation)
+			}
```
[the remark this answers is not in the document]
Comment (mcowger) [reply llm/transformer/openai/aggregator.go:227]: The upstream APIs do not enforce this either. We should not change this behavior.

**Review by looplj — APPROVED** [review]
